### PR TITLE
Relation editor widget UI/UX improvements

### PR DESCRIPTION
**Pull request by nirvn**
Aimed at fixing #1850 .

Here's how it looks in real style:
image

Edit style:
image

I've used a custom scroll bar and our QField green to have it pop out a bit more.

### CODE REVIEW

**Comment by nirvn** [comment]
@3nids , the list is now sorted, very good idea.

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 033bececf45bf1eb5a5b1c7ac848fb64e607889d:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-b9ba2bd515a00fdf75eea63837fbaaff84d3ab3a-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-b9ba2bd515a00fdf75eea63837fbaaff84d3ab3a-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-b9ba2bd515a00fdf75eea63837fbaaff84d3ab3a-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-b9ba2bd515a00fdf75eea63837fbaaff84d3ab3a-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for f38dac731d407daae1bccd8ad5845b9f8b548678:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-06f4676844b0830165475890100c2b77f1273a45-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-06f4676844b0830165475890100c2b77f1273a45-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-06f4676844b0830165475890100c2b77f1273a45-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-06f4676844b0830165475890100c2b77f1273a45-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 4843fb53c1ecdf01c0a8c5c83d8d713cb8530d7b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-35ceda995c522754497a0b22070bb7d16131e6ac-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-35ceda995c522754497a0b22070bb7d16131e6ac-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-35ceda995c522754497a0b22070bb7d16131e6ac-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-35ceda995c522754497a0b22070bb7d16131e6ac-x86.apk)

**Comment by nirvn** [comment]
Last improvement: adding a child feature will result in the relation editor list to re-center its view onto the newly inserted item.

@3nids , that takes care of the last item in your list of improvements.

File: src/qml/editorwidgets/RelationEditor.qml [hunk 1]
```diff
@@ -36,18 +36,38 @@ EditorWidgetBase {
         currentRelationId: relationId
         currentNmRelationId: nmRelationId
         feature: currentFeature
+
+        property int featureFocus: -1
+        onModelUpdated: {
+          if (featureFocus > -1) {
+            console.log(relationEditorModel.getFeatureIdRow(featureFocus))
```
Comment (3nids) [inline src/qml/editorwidgets/RelationEditor.qml:43]: left over?

File: src/qml/editorwidgets/RelationEditor.qml [hunk 2]
```diff
@@ -36,18 +36,38 @@ EditorWidgetBase {
...
+        }
     }
 
     //the list
     ListView {
         id: referencingFeatureListView
         model: relationEditorModel
         width: parent.width
-        height: Math.min( 5 * itemHeight, referencingFeatureListView.count * itemHeight )
+        height: Math.min( 4 * itemHeight, referencingFeatureListView.count * itemHeight ) + ( referencingFeatureListView.count > 0 ? itemHeight / 2 : 0 )
```
Comment (3nids) [inline src/qml/editorwidgets/RelationEditor.qml:55]:
```suggestion
        height: Math.min( 4 * itemHeight, referencingFeatureListView.count * itemHeight ) + ( referencingFeatureListView.count > 4 ? itemHeight / 2 : 0 )
```
?

**Review by 3nids — APPROVED** [review]

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for c7b03bfaf9a636d76a10325dd6eb2c689b35dd3b:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-02c607d5b23f2995596387e9457952bbd3def847-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-02c607d5b23f2995596387e9457952bbd3def847-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-02c607d5b23f2995596387e9457952bbd3def847-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-02c607d5b23f2995596387e9457952bbd3def847-x86.apk)

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 6f9e2ecb18fb125fd80493eae97384409d68e9af:
  - [**arm64_v8a**](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-ac4af34ad4c033a768f528807e42a0fe17559a26-arm64_v8a.apk)

Other architectures: [armv7](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-ac4af34ad4c033a768f528807e42a0fe17559a26-armv7.apk), [x86_64](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-ac4af34ad4c033a768f528807e42a0fe17559a26-x86_64.apk), [x86](https://sos-ch-dk-2.exo.io/qfieldapks/ci-builds/qfield-dev-1857-ac4af34ad4c033a768f528807e42a0fe17559a26-x86.apk)

**Comment by blacktesta** [comment]
Hi guys. I'm sorry if it is wrong to post here. This improvements is what i was thinking about during my (Q)field work. I just want to say when attribute form is maximized it still clipped to 4.5. There is a lot of free unused space left.(https://user-images.githubusercontent.com/74047515/115448720-b5589000-a222-11eb-8c30-bbf465e8c133.jpg)

**Pull request merged**